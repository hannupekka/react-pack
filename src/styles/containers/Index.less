@import '~styles/main';

// Animations
.enter {
  animation: fadeInDown @duration--enter;
  opacity: 0.01;
}

.enter--active {
  opacity: 1;
  transition: all @duration--enter ease-in;
}

.leave {
  animation: fadeOutUp @duration--enter;
  opacity: 1;
}

.leave--active {
  opacity: 0.01;
  transition: opacity @duration--leave ease-out;
}
