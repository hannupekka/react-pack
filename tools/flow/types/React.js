declare type ElementType = React$Element<any>
