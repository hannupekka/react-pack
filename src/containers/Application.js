// @flow
import styles from 'styles/containers/Application.less';
import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import Header from 'components/Header';

type Props = {
  children: ElementType | Array<ElementType>
}

// eslint-disable-next-line react/prefer-stateless-function
class Application extends Component {
  props: Props;
  render(): ElementType {
    const { children } = this.props;

    return (
      <div>
        <Header />
        <section styleName="content">
          {children}
        </section>
      </div>
    );
  }
}

export default CSSModules(Application, styles);
