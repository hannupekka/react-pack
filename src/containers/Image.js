import styles from 'styles/image.less';
import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import ImmutablePropTypes from 'react-immutable-proptypes';
import CSSModules from 'react-css-modules';
import * as ImageActions from 'actions/image';
import Loader from 'components/loader';
import Error from 'components/Error';

class Image extends Component {
  render() {
    const { isLoading, isError, image, actions } = this.props;

    if (isLoading) {
      return <Loader />;
    }

    if (isError) {
      return <Error message="Error requesting image!" />;
    }

    return (
      <div styleName="image">
        <button onClick={() => actions.requestImage()}>Get random image</button>
        {image.get('image_url') && <img src={image.get('image_url')} />}
      </div>
    );
  }
}

Image.propTypes = {
  actions: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  image: ImmutablePropTypes.map,
  params: PropTypes.object // Inserted by react-router-redux
};

const select = store => {
  return {
    isLoading: store.image.get('isLoading'),
    isError: store.image.get('isError'),
    image: store.image.get('image')
  };
};

const mapActions = dispatch => {
  return {
    actions: bindActionCreators(ImageActions, dispatch)
  };
};

export default connect(
  select,
  mapActions
)(CSSModules(Image, styles));
