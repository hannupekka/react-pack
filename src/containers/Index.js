// @flow
import styles from 'styles/containers/Index.less';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
import CSSModules from 'react-css-modules';
import * as uiActions from 'redux/modules/ui';

type Props = {
  showGreeting: boolean,
  dispatch: Function
};

export class Index extends Component {
  props: Props;

  onNavigateToRepoSearch = () => {
    const { dispatch } = this.props;
    dispatch(push('/repos'));
  }

  onToggleGreeting = (): void => {
    const { dispatch } = this.props;
    dispatch(uiActions.toggleGreeting());
  }

  renderGreeting = (): ?React$Element<any> => {
    const { showGreeting } = this.props;

    if (!showGreeting) {
      return null;
    }

    return <h1>Hello world!</h1>;
  }

  render() {
    return (
      <div>
        <button
          styleName="button"
          onClick={this.onNavigateToRepoSearch}
        >Navigate to other page
        </button>
        <button
          styleName="button"
          onClick={this.onToggleGreeting}
        >Toggle greeting
        </button>
        <CSSTransitionGroup
          transitionEnterTimeout={150}
          transitionLeaveTimeout={150}
          transitionName={{
            enter: styles.enter,
            enterActive: styles['enter--active'],
            leave: styles.leave,
            leaveActive: styles['leave--active'],
          }}
        >
          {this.renderGreeting()}
        </CSSTransitionGroup>
      </div>
    );
  }
}

type MappedState = {
  showGreeting: boolean
}

const mapState: Function = (state: State): MappedState => ({
  showGreeting: state.ui.get('showGreeting')
});

export default connect(
  mapState
)(CSSModules(Index, styles));
