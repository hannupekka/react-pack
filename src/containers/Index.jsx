// @flow
import styles from 'styles/containers/Index.less';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import CSSModules from 'react-css-modules';
import * as uiActions from 'redux/ui';
import * as userActions from 'redux/users';

type Props = {
  showGreeting: boolean,
  users: Array<Object>,
  dispatch: Function,
};

export class Index extends Component<Props> {
  onNavigateToRepoSearch = (): void => {
    const { dispatch } = this.props;
    dispatch(push('/repos'));
  };

  onToggleGreeting = (): void => {
    const { dispatch } = this.props;
    dispatch(uiActions.toggleGreeting());
  };

  renderGreeting = (): ?React$Element<any> => {
    const { showGreeting } = this.props;

    if (!showGreeting) {
      return null;
    }

    return <h1>Hello world!</h1>;
  };

  renderUsers = (): ?React$Element<any> => {
    const { users } = this.props;

    if (users.length === 0) {
      return null;
    }

    return (
      <div>
        {users.map((user, i) => (
          <p key={user.email}>
            User {i + 1}: {user.email}
          </p>
        ))}
      </div>
    );
  };

  onFetchRandomUsers = (): void => {
    const { dispatch } = this.props;
    dispatch(userActions.fetchRandomUser());
  };

  render() {
    return (
      <div>
        <button styleName="button" onClick={this.onNavigateToRepoSearch}>
          Navigate to other page
        </button>
        <button styleName="button" onClick={this.onToggleGreeting}>
          Toggle greeting
        </button>
        <button styleName="button" onClick={this.onFetchRandomUsers}>
          Fetch 2 random users
        </button>
        {this.renderGreeting()}
        {this.renderUsers()}
      </div>
    );
  }
}

type MappedState = {
  showGreeting: boolean,
  users: Array<Object>,
};

const mapState: Function = (state: RootState): MappedState => ({
  showGreeting: state.ui.showGreeting,
  users: state.users.users,
});

export default connect(mapState)(CSSModules(Index, styles));
