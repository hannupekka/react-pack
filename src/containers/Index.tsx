import styles from '@app/styles/containers/Index.less';
import * as React from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import map from 'lodash-es/map';
import { createStructuredSelector } from 'reselect';
import * as uiActions from '@app/redux/ui';
import * as userActions from '@app/redux/users';
import { TDispatch, IUser } from '@app/types';
import getUsers from '@app/redux/users/selectors';
import getShowGreeting from '@app/redux/ui/selectors';

interface Props {
  dispatch: TDispatch;
  showGreeting: boolean;
  users: Array<IUser>;
}

export class Index extends React.Component<Props> {
  onNavigateToRepoSearch = () => {
    const { dispatch } = this.props;
    dispatch(push('/repos'));
  };

  onToggleGreeting = () => {
    const { dispatch } = this.props;
    dispatch(uiActions.toggleGreeting());
  };

  renderGreeting = () => {
    const { showGreeting } = this.props;

    if (!showGreeting) {
      return null;
    }

    return <h1>Hello world!</h1>;
  };

  renderUsers = () => {
    const { users } = this.props;

    if (Object.keys(users).length === 0) {
      return null;
    }

    return (
      <div>
        {map(users, (user: IUser, i: number) => (
          <p key={user.email}>
            User {i + 1}: {user.email}
          </p>
        ))}
      </div>
    );
  };

  onFetchRandomUsers = () => {
    const { dispatch } = this.props;
    dispatch(userActions.fetchRandomUser());
  };

  render(): JSX.Element {
    return (
      <div>
        <button className={styles.button} onClick={this.onNavigateToRepoSearch}>
          Navigate to other page
        </button>
        <button className={styles.button} onClick={this.onToggleGreeting}>
          Toggle greeting
        </button>
        <button className={styles.button} onClick={this.onFetchRandomUsers}>
          Fetch 2 random users
        </button>
        {this.renderGreeting()}
        {this.renderUsers()}
      </div>
    );
  }
}

const mapState = createStructuredSelector({
  showGreeting: getShowGreeting,
  users: getUsers,
});

export default connect(mapState)(Index);
