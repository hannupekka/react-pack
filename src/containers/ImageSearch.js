// @flow

import styles from 'styles/containers/ImageSearch';
import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import ImmutablePropTypes from 'react-immutable-proptypes';
import CSSModules from 'react-css-modules';
import { onlyUpdateForKeys } from 'recompose';
import * as imageActions from 'redux/modules/image';
import Loader from 'components/Loader';
import Error from 'components/Error';
import Image from 'components/Image';

class ImageSearch extends Component {
  constructor(props) {
    super(props);

    (this:any).bindSearch = (c) => ((this:any).search = c);
  }

  onFetchImage = () => {
    const search = (this:any).search.value;
    this.props.fetchImage(search);
  }

  render() {
    const { isLoading, isError, image } = this.props;

    if (isError) {
      return <Error message="Error requesting image!" />;
    }

    return (
      <div styleName="image-search">
        <label htmlFor="search">Search word</label>
        <input type="text" ref={(this:any).bindSearch} />
        <button onClick={this.onFetchImage}>
          Get random image <i className="fa fa-search"></i>
        </button>
        {isLoading && <Loader />}
        <Image src={image.get('fixed_height_downsampled_url')} />
      </div>
    );
  }
}

ImageSearch.propTypes = {
  fetchImage: PropTypes.func.isRequired,
  params: PropTypes.object,
  image: ImmutablePropTypes.map.isRequired,
  isError: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const select = store => ({
  image: store.image.get('image'),
  isError: store.image.get('isError'),
  isLoading: store.image.get('isLoading')
});

const mapActions = dispatch => {
  return {
    fetchImage: tag => dispatch(imageActions.fetchImage(tag))
  };
};

export default connect(
  select,
  mapActions
)(onlyUpdateForKeys(['isLoading', 'isError', 'image'])(CSSModules(ImageSearch, styles)));
