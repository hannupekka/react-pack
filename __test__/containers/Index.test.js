import React from 'react';
import renderer from 'react-test-renderer';
import { Index } from 'containers/Index';

describe('Index', () => {
  it('renders correctly without greeting', () => {
    const tree = renderer.create(
      <Index
        showGreeting={false}
        toggleGreeting={() => {}}
      />
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with greeting', () => {
    const tree = renderer.create(
      <Index
        showGreeting
        toggleGreeting={() => {}}
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
