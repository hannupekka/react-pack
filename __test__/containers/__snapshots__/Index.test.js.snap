// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly with greeting 1`] = `
<div>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Navigate to other page
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Toggle greeting
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Fetch 2 random users
  </button>
  <h1>
    Hello world!
  </h1>
</div>
`;

exports[`Index renders correctly with users 1`] = `
<div>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Navigate to other page
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Toggle greeting
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Fetch 2 random users
  </button>
  <div>
    <p>
      User 
      1
      : 
      foo.bar@example.com
    </p>
    <p>
      User 
      2
      : 
      bar.foo@example.com
    </p>
  </div>
</div>
`;

exports[`Index renders correctly without greeting 1`] = `
<div>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Navigate to other page
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Toggle greeting
  </button>
  <button
    onClick={[Function]}
    styleName="button"
  >
    Fetch 2 random users
  </button>
</div>
`;
